llvm/lib/Target/NVPTX/NVPTX.h

```diff
@@ -106,15 +106,53 @@ enum LoadStore {
   isStoreShift = 6
 };
 
-namespace PTXLdStInstCode {
-enum MemorySemantic {
+// Extends LLVM AtomicOrdering with PTX Orderings:
+using OrderingUnderlyingType = unsigned int;
+enum class Ordering : OrderingUnderlyingType {
   NotAtomic = 0, // PTX calls these: "Weak"
-  Volatile = 1,
+  // Unordered = 1, // NVPTX maps LLVM Unorderd to Relaxed
   Relaxed = 2,
-  Acquire = 3,
-  Release = 4,
-  RelaxedMMIO = 5
+  // Consume = 3,   // Unimplemented in LLVM; NVPTX would map to "Acquire"
+  Acquire = 4,
+  Release = 5,
+  // AcquireRelease = 6, // TODO
+  SequentiallyConsistent = 7,
+  Volatile = 8,
+  RelaxedMMIO = 9,
+  LAST = RelaxedMMIO
 };
+
+template <typename OStream> OStream &operator<<(OStream &O, Ordering Order) {
```

Comment:
Good idea. I've done that split into a "to c string" (`char const*`) conversion plus using `raw_ostream`.
These are inline functions here for now, but we could move their implementation to a .cpp file. Itried a few of them, and couldn't find one that still allows everything to link. Do you have a suggestion for which .cpp file to put these in? Or should I just leave them as inline functions here for now?

Reply:
Moving them to NVPTXUtility.* would probably make sense.

Reply:
I tried that but it failed to link throughout. I think I'd need to increase the set of TUs that link against it. Would it be possible for me to do it in a separate PR? 

The next PR in the queue is a NFC refactor, so it would fit in there.

Reply:
Fixed that here, resolving.